src/modules/Dropdown/Dropdown.js

```diff
@@ -1201,10 +1214,14 @@ export default class Dropdown extends Component {
   }
 
   renderOptions = () => {
-    const { multiple, search, noResultsMessage } = this.props
+    const { maxSelections, multiple, search, noResultsMessage } = this.props
     const { selectedIndex, value } = this.state
     const options = this.getMenuOptions()
 
+    if (!_.isNil(maxSelections) && value.length >= maxSelections) {
+      return <div className='message'>Max {maxSelections} selection{maxSelections > 1 && 's'}</div>
```

Comment:
Let's create a new subcomponent `DropdownMessage`. This will allow to use factories for it, see `DropdownSearchInput` for reference.

It will accept `type` (`maxSelections` or `noResultsMessage`) and `value` (`maxSelections`). This will allow our users customize message better and localize it, too.